apollo-router/src/cache/redis.rs

```diff
@@ -332,6 +333,14 @@ impl RedisCacheStorage {
             .build_pool(pool_size)?;
 
         for client in pooled_client.clients() {
+            // Use a client for the replicas if in clustered mode, otherwise we'll only connect to
+            // the writer nodes in the cluster
+            let client = if is_cluster {
+                &client.replicas().client()
```

Comment:
Note `.client()` returns the inner non-replica client again, so it doesn't change anything this way.

We need to use `.replicas()` wherever we execute Redis commands. I _think_ it's fine to do `.replicas()` even if there are no replicas, but we might want to validate that :)

Reply:
oh, yeah, you're right:

```
  /// Read the underlying [RedisClient](crate::clients::Client) that interacts with primary nodes.
  pub fn client(&self) -> Client {
    Client::from(self.client.inner())
  }
```